packages/create-llama/templates/components/engine/agent/tools.py

```diff
@@ -0,0 +1,59 @@
+import os
+import json
+import importlib
+
+
+from typing import Union
+from pydantic import BaseModel, field_validator, computed_field
+from llama_index.tools.types import BaseTool
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from llama_index.agent import OpenAIAgent, ReActAgent
+from llama_index.tools.query_engine import QueryEngineTool
+
+
+class ToolConfig(BaseModel):
+    name: str
+    raw_config: str
+
+    @computed_field
+    def config(self) -> dict:
+        return json.loads(self.raw_config)
+
+    @field_validator("raw_config")
+    @classmethod
+    def config_must_serializable(cls, value) -> str:
+        try:
+            json.loads(value)
+            return value
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Config {value} is not a serializable JSON!")
+
+    @staticmethod
+    def from_env() -> "ToolConfig":
+        return ToolConfig(
+            name=os.environ.get("TOOL", ""),
+            raw_config=os.environ.get("TOOL_CONFIG", ""),
+        )
+
+
+class ToolFactory:
+
+    @staticmethod
+    def create_tool(tool_name: str, **kwargs) -> Union[BaseTool, BaseToolSpec]:
```

Comment:
could return `None` if there are no tools in env?

Reply:
don't you think it better throws errors if the tool is not exist?